From 2cd8a6f3e1b5813faa931077a63c78d33fe5ba69 Mon Sep 17 00:00:00 2001
From: Andrii Osipov <aosipov@anaconda.com>
Date: Wed, 26 Mar 2025 07:26:55 -0700
Subject: [PATCH] remove @ from path

---
 cmake/nanobind-config.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/nanobind-config.cmake b/cmake/nanobind-config.cmake
index cc11fb9..f31bd0a 100644
--- a/cmake/nanobind-config.cmake
+++ b/cmake/nanobind-config.cmake
@@ -95,7 +95,7 @@ function (nanobind_link_options name)
     else()
       set(NB_LINKER_RESPONSE_FILE darwin-ld-cpython.sym)
     endif()
-    target_link_options(${name} PRIVATE "-Wl,@${NB_DIR}/cmake/${NB_LINKER_RESPONSE_FILE}")
+    target_link_options(${name} PRIVATE "-Wl,${NB_DIR}/cmake/${NB_LINKER_RESPONSE_FILE}")
   endif()
 endfunction()
 
-- 
2.39.3 (Apple Git-146)

