From 948920f5421ba1eded76e06cca95862bf92c15fe Mon Sep 17 00:00:00 2001
From: Andrii Osipov <aosipov@anaconda.com>
Date: Wed, 26 Mar 2025 06:02:50 -0700
Subject: [PATCH] use response file for symbol list and enable dynamic lookup

---
 cmake/nanobind-config.cmake | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/cmake/nanobind-config.cmake b/cmake/nanobind-config.cmake
index cc11fb9..aebcc70 100644
--- a/cmake/nanobind-config.cmake
+++ b/cmake/nanobind-config.cmake
@@ -95,7 +95,10 @@ function (nanobind_link_options name)
     else()
       set(NB_LINKER_RESPONSE_FILE darwin-ld-cpython.sym)
     endif()
-    target_link_options(${name} PRIVATE "-Wl,@${NB_DIR}/cmake/${NB_LINKER_RESPONSE_FILE}")
+    target_link_options(${name} PRIVATE
+        "-Wl,@${NB_DIR}/cmake/${NB_LINKER_RESPONSE_FILE}"
+        "-undefined" "dynamic_lookup"
+    )
   endif()
 endfunction()
 
-- 
2.39.3 (Apple Git-146)

